parsers/CA_AB.py

```diff
@@ -115,48 +129,26 @@ def fetch_production(
         f"{URL_STRING}/CSDReportServlet", params={"contentType": "csv"}
     )
     generation = {
-        row[0]: {
-            "MC": float(row[1]),  # maximum capability
-            "TNG": float(row[2]),  # total net generation
-        }
+        row[0]: float(row[2])  # total net generation
         for row in csv.reader(response.text.split("\r\n\r\n")[3].splitlines())
     }
-    return [
-        validation.validate(
-            {
-                "capacity": {
-                    "gas": generation["COGENERATION"]["MC"]
-                    + generation["COMBINED CYCLE"]["MC"]
-                    + generation["GAS FIRED STEAM"]["MC"]
-                    + generation["SIMPLE CYCLE"]["MC"],
-                    "wind": generation["WIND"]["MC"],
-                    "solar": generation["SOLAR"]["MC"],
-                    "hydro": generation["HYDRO"]["MC"],
-                    "biomass": generation["OTHER"]["MC"],
-                    "battery storage": generation["ENERGY STORAGE"]["MC"],
-                },
-                "datetime": get_csd_report_timestamp(response.text),
-                "production": {
-                    "gas": generation["COGENERATION"]["TNG"]
-                    + generation["COMBINED CYCLE"]["TNG"]
-                    + generation["GAS FIRED STEAM"]["TNG"]
-                    + generation["SIMPLE CYCLE"]["TNG"],
-                    "wind": generation["WIND"]["TNG"],
-                    "solar": generation["SOLAR"]["TNG"],
-                    "hydro": generation["HYDRO"]["TNG"],
-                    "biomass": generation["OTHER"]["TNG"],
-                },
-                "source": URL.netloc,
-                "storage": {
-                    "battery": generation["ENERGY STORAGE"]["TNG"],
-                },
-                "zoneKey": zone_key,
-            },
-            logger,
-            floor=MINIMUM_PRODUCTION_THRESHOLD,
-            remove_negative=True,
-        )
-    ]
+
+    production_breakdowns = ProductionBreakdownList(logger)
+    mix = ProductionMix()
+    for key in generation:
+        if key in PRODUCTION_MAPPING:
+            mix.add_value(PRODUCTION_MAPPING[key], generation.get(key))
```

Comment:
Maybe add some waning logs here if the key is not in the production mapping (or a key that should be skipped) so we know if they add any new keys in the future.